src/django_components/tag_formatter.py

```diff
@@ -0,0 +1,269 @@
+import abc
+import re
+from typing import List, Tuple, Type
+
+from django.template import TemplateSyntaxError
+from django.utils.module_loading import import_string
+
+from django_components.app_settings import app_settings
+from django_components.expression import resolve_string
+from django_components.template_parser import VAR_CHARS
+from django_components.utils import is_str_wrapped_in_quotes
+
+TAG_RE = re.compile(r"^[{chars}]+$".format(chars=VAR_CHARS))
+
+
+class TagFormatterABC(abc.ABC):
```

Comment:
This is the API of the `TagFormatter`, inspired by https://github.com/Xzya/django-web-components#component-tag-formatter. To define a custom `TagFormatter`, one must define following functions:

- `format_block_start_tag`
- `format_block_end_tag`
- `format_inline_tag`
- `parse_block_start_tag`
- `parse_inline_tag`

Unlike `django-web-components`, our `TagFormatter` requires that we define also how to extract (parse) the component name from the inputs. This was required so we could transform our current way of defining components into a TagFormatter.
```django
{% component "my_comp" %}
```
Because `django-web-components` assumed that the component name can be extracted from the tag name. But in our case, the tag name is `"component"`, so on its own it's not enough information. So instead we pass all the args/kwargs to the parser, which would extract the name `my_comp` from the example above, and return other args/kwargs unchanged.